calculator/main.py

```diff
@@ -58,18 +103,27 @@ def get_history(calc):
         ie:
         ('2016-05-20 12:00:00', 'add', (1, 2), 3),
     """
-    pass
+    return calc['history']
 
 
 def reset_history(calc):
     """
     Resets the calculator history back to an empty list.
     """
-    pass
+    calc['history'] = []
 
 
 def repeat_last_operation(calc):
     """
     Returns the result of the last operation executed in the history.
     """
-    pass
+    # TODO: catch exceptions if history is nonexistant.
+    
+    if len(calc['history']) == 0:
+        return None
+    else:    
+        # Can break this into separate lines to extract.
+        # Or you can make one really long complex hard to read line.
+        # or just discard the unused values with the commonly used _.
+        _, operation, params, _ = calc['history'][-1]
+        return perform_operation(calc, operation, params)
```

Comment:
From what Elliot told me, I believe python normally returns None if it hasn't been told to return anything else, so you instead check if its not empty, if so do something, and if its empty it'd just return None anyway.